tests/internal/cluster/cluster.go

```diff
@@ -171,3 +179,209 @@ func ExecCmdWithStdout(
 
 	return outputMap, nil
 }
+
+// new commands
+
+// ExecLocalCommand runs the provided command with the provided args locally, cancelling execution if it exceeds
+// timeout.
+func ExecLocalCommand(timeout time.Duration, command string, args ...string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+	defer cancel()
+
+	glog.V(params.LogLevel).Infof("Locally executing command '%s' with args '%v'", command, args)
+
+	output, err := exec.CommandContext(ctx, command, args...).Output()
+
+	return string(output), err
+}
+
+// ExecCmdWithRetries executes a command on the provided client on each node matching nodeSelector,
+// retrying on internal errors
+// retries times with a 10 second delay between retries, and ignores the stdout.
+func ExecCmdWithRetries(client *clients.Settings, retries uint, nodeSelector, command string) error {
+	for retry := range retries {
+		err := ExecCmd(client, nodeSelector, command)
+		if isErrorExecuting(err) {
+			glog.V(params.LogLevel).Infof("Error during command execution, retry %d (%d max): %w", retry+1, retries, err)
+
+			time.Sleep(10 * time.Second)
+
+			continue
+		}
+
+		return err
+	}
+
+	return fmt.Errorf("ran out of %d retries executing command %s", retries, command)
+}
+
+// ExecCmdWithStdoutWithRetries executes a command on the provided client,
+// retrying on internal errors retries times with a 10
+// second delay between retries, and returns the stdout for each node.
+func ExecCmdWithStdoutWithRetries(
+	client *clients.Settings, retries uint, command string, options ...metav1.ListOptions) (map[string]string, error) {
+	for retry := range retries {
+		outputs, err := ExecCmdWithStdout(client, command, options...)
+		if isErrorExecuting(err) {
+			glog.V(params.LogLevel).Infof("Error during command execution, retry %d (%d max): %w", retry+1, retries, err)
+
+			time.Sleep(10 * time.Second)
+
+			continue
+		}
+
+		return outputs, err
+	}
+
+	return nil, fmt.Errorf("ran out of %d retries executing command %s", retries, command)
+}
+
+// ExecCommandOnSNOWithRetries executes a command on the provided single node client,
+// retrying on internal errors retries times
+// with a 10 second delay between retries, and returns the stdout.
+func ExecCommandOnSNOWithRetries(client *clients.Settings, retries uint, command string) (string, error) {
+	outputs, err := ExecCmdWithStdoutWithRetries(client, retries, command)
+	if err != nil {
+		return "", err
+	}
+
+	if len(outputs) != 1 {
+		return "", fmt.Errorf("expected results from one node, found %d nodes", len(outputs))
+	}
+
+	for _, output := range outputs {
+		return output, nil
+	}
+
+	return "", fmt.Errorf("found unreachable code in ExecCommandOnSNO")
+}
+
+// isErrorExecuting matches errors that contain the message "error executing command in container".
+func isErrorExecuting(err error) bool {
```

Comment:
Please move private function to the end of the file.

Reply:
fixed